mantidproject/mantid
# Add histogram specific integration ranges to Integration

This PR adds two new properties to the [`Integration`](http://docs.mantidproject.org/nightly/algorithms/Integration-v1.html) algorithm: *RangeLowerList* and *RangeUpperList* can be used to give integration limits to each spectrum individually. This will help, for example, the data reduction at ILL on spectrometers where the L2 distance and thus the elastic peak position in TOF axis is different for each detector.

**To test:**

Try out different combinations of parameters, also *IncludePartialBins*. Test that there are no hidden surprises and that the algorithm fails gracefully if upper range is smaller than the lower range. Check that the X values correspond to the integration limits. Try out ragged workspaces, if possible.

Fixes #18657.

*Release notes were updated accordingly.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
